statesync/reactor_test.go

```diff
@@ -1,21 +1,85 @@
 package statesync
 
 import (
+	"context"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
 	"github.com/tendermint/tendermint/p2p"
-	p2pmocks "github.com/tendermint/tendermint/p2p/mocks"
 	ssproto "github.com/tendermint/tendermint/proto/tendermint/statesync"
 	proxymocks "github.com/tendermint/tendermint/proxy/mocks"
 )
 
-func TestReactor_Receive_ChunkRequest(t *testing.T) {
+type ReactorTestSuite struct {
+	suite.Suite
+
+	ctx            context.Context
+	cancel         context.CancelFunc
+	reactor        *Reactor
+	snapshotChShim *p2p.ChannelShim
+	chunkChShim    *p2p.ChannelShim
+	peerUpdateCh   chan p2p.PeerUpdate
+}
+
+// SetupTest is executed per test and creates and starts a new state sync reactor.
+func (rts *ReactorTestSuite) SetupTest() {
+	conn := &proxymocks.AppConnSnapshot{}
```

Comment:
If we store this as `rts.conn` then we won't need `setConn()`.

Reply:
I removed `setConn`. We shouldn't need it anymore.